sdk/ai/Azure.AI.Inference/samples/Sample8_ChatCompletionsWithTelemetry.md

```diff
@@ -0,0 +1,145 @@
+# Using telemetry with Azure.AI.Inference
+
+In this example we will demonstrate how [OpenTelemetry](https://learn.microsoft.com/dotnet/core/diagnostics/observability-with-otel) library can be leveraged to diagnose the issues in the Chat completion request.
+
+## Project creation and dependency installation.
+First, we will create the console application project and add `Azure.AI.Inference` as a dependency. The first command will create the project called `TelemetryDemo.csproj`. The ` dotnet add package <…>` command will modify project file and in future we will need to run `dotnet restore` to install all dependencies, if we will remove the installed ones.
+
+```bash
+dotnet new console --name TelemetryDemo --output TelemetryDemo
+dotnet add package Azure.AI.Inference --prerelease
+```
+
+Now we will need to add the dependencies for OpenTelemetry.
+
+```bash
+dotnet add package OpenTelemetry
+dotnet add package OpenTelemetry.Exporter.Console
+dotnet add package Azure.Monitor.OpenTelemetry.AspNetCore --prerelease
+```
+
+## Create the simple application with telemetry.
+The `dotnet new` created the project with the single file called Program.cs. Let us edit this file with the IDE of choice.
+
+First we will import open telemetry and Azure.AI.Inference
+```c#
+//Azure imports
+using Azure;
+using Azure.AI.Inference;
+// Open telemetry imports
+using OpenTelemetry;
+using OpenTelemetry.Resources;
+using OpenTelemetry.Trace;
+using Azure.Monitor.OpenTelemetry.Exporter;
+using OpenTelemetry.Metrics;
+```
+
+Now we will define the `endpoint`, `credential` and `model`. In this example we will take these values from the environment variables.
+```c#
+var endpoint = new Uri(System.Environment.GetEnvironmentVariable("MODEL_ENDPOINT"));
+var credential = new AzureKeyCredential(System.Environment.GetEnvironmentVariable("GITHUB_TOKEN"));
+var model = System.Environment.GetEnvironmentVariable("MODEL_NAME");
+```
+
+Next, we will define the constant, which defines, what [Activity](https://learn.microsoft.com/dotnet/api/system.diagnostics.activity) and [Meter](https://learn.microsoft.com/dotnet/api/system.diagnostics.metrics.meter) do we want to listen to. In Azure.AI.Inferencecase they are both named `Azure.AI.Inference.ChatCompletionsClient`.
+
+```c#
+const string ACTIVITY = "Azure.AI.Inference.ChatCompletionsClient";
+```
+
+To log metrics and events to Application Insights we will need to get the connection string. Please open the Azure portal, create the Application Insights resource you wish to use for storing the telemetry. After that open the main page and find the "Connection String". It generally will have the format similar to "InstrumentationKey=xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx;IngestionEndpoint=https://region-x.in.applicationinsights.azure.com/;LiveEndpoint=https://eastus.livediagnostics.monitor.azure.com/;ApplicationId=xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx". In our code we will create the constant, storing the connection string.
```

Comment:
@scottaddie :  Should we be adding new samples that are based on connection strings, or should be demonstrating identity credentials for all the things?

Reply:
removed, AzMon distro happy use-case and main scenario use connections string provided as environment variable. Also, the default way to authenticate to appinsights is via connection strings